h/views/account_signup.py

```diff
@@ -280,8 +400,11 @@ def idinfo_jwt_decode_error(self):
     @exception_view_config(context=ValidationFailure, route_name="signup.orcid")
     @exception_view_config(context=ValidationFailure, route_name="signup.google")
     @exception_view_config(context=ValidationFailure, route_name="signup.facebook")
+    @exception_view_config(context=UsernameConflictError, route_name="signup.orcid")
+    @exception_view_config(context=UsernameConflictError, route_name="signup.google")
+    @exception_view_config(context=UsernameConflictError, route_name="signup.facebook")
```

Comment:
When `POST`'ing the `/signup/<PROVIDER>` form with a username that's already taken the `post()` view will raise `ValidationFailure` and that was already handled here.

What was not handled was when validation succeeds and then a concurrent request creates a user with a conflicting username and when the first request calls `UserSignupService.signup()` it raises `UsernameConflictError`. Added handling of that to this exception view since the error experience that we want is the same as for a validation failure.